File: EasyCard/Transactions.Business/Entities/BillingSchedule.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Text;
using Transactions.Shared.Enums;

namespace Transactions.Business.Entities
{
    public class BillingSchedule
    {
        public RepeatPeriodTypeEnum RepeatPeriodType { get; set; }

        public JObject RepeatPeriod { get; set; }

        public StartAtTypeEnum StartAtType { get; set; }

        public JObject StartAt { get; set; }

        public EndAtTypeEnum EndAtType { get; set; }

        public JObject EndAt { get; set; }
    }
}

File: EasyCard/UI/merchant/src/extensions/filters/billingschedule.js
export default function(value){
    if(value == null || !value.repeatPeriodType){
        return null;
    }
    let str = `${value.repeatPeriodType}: ${value.repeatPeriod || '-'}`;
    if(value.startAtType){
        str += ` / ${value.startAtType}: ${value.startAt}`;
    }
    if(value.endAtType){
        str += ` / ${value.endAtType}: ${value.endAt}`;
    }
    if(value.endAtNumberOfPayments){
        str += ` | ${value.endAtNumberOfPayments}`;
    }

    return str;
}